src/wayland/buffer.rs

```diff
@@ -15,6 +14,49 @@ use wayland_client::{
 
 use super::State;
 
+#[cfg(any(target_os = "linux", target_os = "freebsd"))]
+fn create_memfile() -> File {
+    use nix::sys::memfd::{memfd_create, MemFdCreateFlag};
+
+    let name = unsafe { CStr::from_bytes_with_nul_unchecked("softbuffer\0".as_bytes()) };
+    let fd = memfd_create(name, MemFdCreateFlag::MFD_CLOEXEC)
```

Comment:
We should fallback to shm files if creating a memfd failed, as we could get `ENOTSUPP` on older versions of Linux.

Reply:
I think it will fail to compile anyway if libc doesn't have the function, so this would only help with a libc version that has the function and a kernel that doesn't?

There's also FreeBSD's `SHM_ANON` argument to `shm_open` which is useful here on older versions without `memfd_create`. But not useful if it doesn't compile anyway.

Reply:
Including the fallback on Linux/FreeBSD also means depending on fastrand there. Which isn't a big deal, but seems unnecessary if it's going to just fail to compile where this code path would be used.

Reply:
I agree that if it links to the function it is pribably implemented, either by the kernel or as a shim.